middle_end/flambda2/types/env/typing_env.ml

```diff
@@ -875,15 +845,13 @@ and add_equation1 t name ty ~(meet_type : meet_type) =
       let[@inline always] name eqn_name ~coercion =
         assert (Coercion.is_id coercion);
         (* true by definition *)
-        if Name.equal name eqn_name
```

Comment:
Yes! The lack of a meet was causing us to lose equations. We couldn't remove the shortcut until we had a way to prevent infinite loops caused by meet re-discovering existing equations, but the current PR provides that and we can do the correct thing all the time.

Reply:
(The comment will indeed need to be updated, I'll do that once I get a bit of time to work on this)

Reply:
Ah yes I see, I had missed we were inside `add_equation1` which is passed a type as well as `name`.